### PR TITLE
Refactor Dockerfile debian and alpine

### PR DESCRIPTION
**Pull request by casell**
Refactoring of debian dockerfile to reduce image size and update of alpine dockerfile to bring it on par with the debian one

### CODE REVIEW

File: opendj-packages/opendj-docker/alpine/Dockerfile [hunk 1]
```diff
@@ -2,31 +2,43 @@ FROM openjdk:8-jre-alpine
...
-ARG LDAPS_PORT=1636
-
-ARG BASE_DN="dc=example,dc=com"
+ARG VERSION=@project_version@
 
-ARG ROOT_USER_DN="cn=Directory Manager"
+WORKDIR /opt
 
-ARG ROOT_PASSWORD=password
+RUN apk add --update --no-cache --virtual builddeps wget unzip \
```
Comment (maximthomas) [inline opendj-packages/opendj-docker/alpine/Dockerfile:21]: Got error while building image
```
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c apk add --update --no-cache --virtual builddeps wget unzip  && apk upgrade --update --no-cache  && apk add bash  && wget --quiet https://github.com/OpenIdentityPlatform/OpenDJ/releases/download/$VERSION/opendj-$VERSION.zip  && unzip opendj-$VERSION.zip  && apk del builddeps  && rm -r opendj-$VERSION.zip  && addgroup -S $OPENDJ_USER  && adduser -S -u 1001 -G $OPENDJ_USER $OPENDJ_USER  && install -d -o $OPENDJ_USER /opt/opendj/data  && chown -R $OPENDJ_USER:$OPENDJ_USER /opt/opendj  && chmod -R g=u /opt/opendj]: exit code: 8
```

Comment (casell) [reply opendj-packages/opendj-docker/alpine/Dockerfile:21]: You'll have to pass the buildarg VERSION (version of opendj) and build from the opendj-docker:

`docker build -t openidentityplatform/opendj:4.4.11-alpine --build-arg VERSION=4.4.11 -f alpine/Dockerfile .`

Comment (maximthomas) [reply opendj-packages/opendj-docker/alpine/Dockerfile:21]: Ok, got it, thank you for the contribution @casell !

**Review by maximthomas — APPROVED** [review]

**Comment by casell** [comment]
@maximthomas do you have any plan to tag/release?

**Comment by maximthomas** [comment]
@casell, yes but I can't tell you specific date
@vharseko FYI

**Pull request merged**